src/devices/Common/System/Device/Analog/TriggerReason.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Analog
+{
+    /// <summary>
+    /// Gives the reason why a new value was provided
+    /// </summary>
+    public enum TriggerReason
+    {
+        /// <summary>
+        /// The reason for the new message is unknown
+        /// </summary>
+        Unknown,
```

Comment:
do we use it anywhere in the current code? I'd rather have us add new values when we need to rather than having generic reason on the list.

Also to me this seems more like Error enum. It's either OK reading or something went wrong.

Reply:
This is considering future implementations of this interface/enum. Some ADCs have configurable interrupt capabilities and will fire an update only when the value changed in a specific way (i.e. it exceeds a certain threshold). Therefore I added this.